services/src/main/java/com/linkedin/kmf/services/configs/ProduceServiceConfig.java

```diff
@@ -49,24 +52,17 @@
   public static final String PRODUCER_PROPS_CONFIG = "produce.producer.props";
   public static final String PRODUCER_PROPS_DOC = "The properties used to config producer in produce service.";
 
-  public static final String AUTO_TOPIC_REPLICATION_FACTOR_CONFIG = "produce.topic.autoTopicReplicationFactor";
-  public static final String AUTO_TOPIC_REPLICATION_FACTOR_DOC = "When a topic is created automatically this is the "
+  public static final String TOPIC_REPLICATION_FACTOR_CONFIG = "topic.replicationFactor";
+  public static final String TOPIC_REPLICATION_FACTOR_DOC = "When a topic is created automatically or rebalanced this is the "
       + "replication factor used.";
 
-  public static final String REBALANCE_ENABLED_CONFIG = "produce.topic.rebalanceEnabled";
-  public static final String REBALANCE_ENABLED_DOC = "Periodically move leaders and replica assignments so they are "
-      + "evenly spread amongst brokers.";
-
-  public static final String REBALANCE_PARTITION_MULTIPLE_CONFIG = "produce.topic.rebalancePartitionMultiple";
-  public static final String REBALANCE_PARTITION_MULTIPLE_DOC = "Determines the number of partitions per broker in the ideal case.";
-
-  public static final String REBALANCE_THRESHOLD_CONFIG = "produce.topic.rebalanceThreshold";
-  public static final String REBALANCE_THRESHOLD_DOC = "Determines the number of partitions per broker in the ideal case.";
+  public static final String PARTITIONS_PER_BROKER_CONFIG = "topic.partitionsPerBroker";
```

Comment:
By convention we prefix the config name and the variable name with the service name, e.g. "produce.producer.props". The goal is to avoid config name collision between different services and make it easy for user to know which service is the config applied to. Therefore we probably want to change the name for "topic.replicationFactor" and "topic.partitionsPerBroker" as appropriate.

Previously we discussed which service should take care of topic creation, ProduceService or TopicRebalanceService. Now I look at the patch, it seems cleaner to do it in the TopicRebalanceService. Otherwise we need to define partitionToBrokerRatio in multiple services, which seems ugly. It seems reasonable to have one module that manages both topic rebalance and topic creation.

It is true that produce service won't be able to run if topic is not created and rebalance service is not used. But if rebalance service is explicitly disabled (suppose we enable it by default), it means user wants to manually manage the number of partitions in case of cluster expansion, which in turn indicates that they are OK with manually creating topic.